File: pixiejsClient/src/app/ecs/components/EffectComponent.ts
import { Component, component, serverField } from "../core/Component";
import { ServerComponentType } from "../../enums/ComponentIds";
import { EffectType } from "../../enums/EffectType";

@component(ServerComponentType.Effect)
export class EffectComponent extends Component {
  @serverField(1, "u8") effectType: EffectType;
  @serverField(2, "u32") color: number;

  constructor(entityId: string, effectType: EffectType = EffectType.Spawn, color: number = 0xffffff) {
    super(entityId);
    this.effectType = effectType;
    this.color = color;
  }
}

File: server/Simulation/Components/EffectComponent.cs
using System.Runtime.InteropServices;
using server.ECS;
using server.Enums;

namespace server.Simulation.Components;

[Component(ComponentType = ComponentType.Effect, NetworkSync = true)]
[StructLayout(LayoutKind.Sequential, Pack = 1)]
public struct EffectComponent(EffectType effectType, uint color = 0xFFFFFF)
{
    public long ChangedTick = NttWorld.Tick;

    public EffectType EffectType = effectType;
    public uint Color = color;
}
